src/generation.rs

```diff
@@ -34,6 +38,19 @@ use std::sync::mpsc;
 const UPDATE_COUNT: i64 = 100_000;
 const MAX_POINTS_PER_NODE: i64 = 100_000;
 
+impl CubeNodeWriter {
+    pub fn from_data_provider(
+        octree_data_provider: &OnDiskOctreeDataProvider,
+        octree_meta: &OctreeMeta,
+        node_id: &NodeId,
+    ) -> Self {
+        let path = octree_data_provider.stem(node_id);
```

Comment:
Shouldn't we factor that out?

Reply:
What exactly? `generation.rs` belongs to octree specific stuff (which will be moved into the `octree` subfolder in my next PR), so I think this can stay here...

Reply:
I just felt like it's a pretty "large" parameter, not in a performance sense, but conceptually. For instance, it could make testing a little easier to be able to just instantiate a CubeNodeWriter with some temp file.

Reply:
On the other hand, I don't know OnDiskOctreeDataProvider very well – if my suggestion would lead to lots of redundancy, maybe it's better to leave the file name generation here.

Reply:
It would also sort of decouple the concept of a Cube with points from an octree & its naming scheme.

Reply:
As discussed offline, we will just keep it as it is, as there still is the `new` method. I made this one private to clarify its use.